src/Libraries/Microsoft.Extensions.Telemetry/Sampling/TraceBasedSampler.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Diagnostics.Sampling;
+internal sealed class TraceBasedSampler : LoggerSampler
+{
+    public override bool ShouldSample(SamplingParameters _) =>
+        Activity.Current?.Recorded ?? false;
```

Comment:
We might want no Activity to be true, or maybe to be configurable via the API. This is a spot where some experimental feedback feels very useful.

Reply:
Updated

Reply:
nit: this can be written like 

`Activity.Current?.Recorded is true`. will invoke `Current` once.

Reply:
Taking into account the @noahfalk comment that we should sample in if there is no Activity, I think this makes sense
`Activity.Current?.Recorded ?? true`